app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationServiceTest.java

```diff
@@ -1098,4 +1098,165 @@ public void validGetApplicationPagesMultiPageApp() {
+        action3.setActionConfiguration(actionConfiguration);
+
+        ActionDTO savedAction3 = layoutActionService.createAction(action3).block();
+
+        ApplicationAccessDTO applicationAccessDTO = new ApplicationAccessDTO();
+        applicationAccessDTO.setPublicAccess(true);
+
+        // Trigger the change view access of application now.
+        applicationService.changeViewAccess(originalApplication.getId(), applicationAccessDTO)
+                .timeout(Duration.ofMillis(10))
+                .subscribe();
+
+        Mono<Application> applicationFromDbPostViewChange = Mono.just(originalApplication)
+                .flatMap(originalApp -> {
+                    try {
+                        // Before fetching the cloned application, sleep for 5 seconds to ensure that the cloning finishes
+                        Thread.sleep(5000);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    return applicationRepository.findById(originalApplication.getId(), READ_APPLICATIONS);
+                })
+                .cache();
+
+        Mono<List<NewAction>> actionsMono = applicationFromDbPostViewChange
+                .flatMap(clonedAppFromDb -> newActionService
+                        .findAllByApplicationIdAndViewMode(clonedAppFromDb.getId(), false, READ_ACTIONS, null)
+                        .collectList()
+                );
+
+        Mono<List<PageDTO>> pagesMono = applicationFromDbPostViewChange
+                .flatMapMany(application -> Flux.fromIterable(application.getPages()))
+                .flatMap(applicationPage -> newPageService.findPageById(applicationPage.getId(), READ_PAGES, false))
+                .collectList();
+
+        Mono<Datasource> datasourceMono = applicationFromDbPostViewChange
+                .flatMap(application -> datasourceService.findById(savedDatasource.getId(), READ_DATASOURCES));
+
+        StepVerifier
+                .create(Mono.zip(applicationFromDbPostViewChange, actionsMono, pagesMono, datasourceMono))
+                .assertNext(tuple -> {
+                    Application updatedApplication = tuple.getT1();
+                    List<NewAction> actions = tuple.getT2();
+                    List<PageDTO> pages = tuple.getT3();
+                    Datasource datasource1 = tuple.getT4();
+
+                    assertThat(updatedApplication).isNotNull();
+                    assertThat(updatedApplication.getIsPublic()).isTrue();
+                    assertThat(updatedApplication
+                            .getPolicies()
+                            .stream()
+                            .filter(policy -> {
+                                if (policy.getPermission().equals(READ_APPLICATIONS.getValue())) {
+                                    return true;
+                                }
+                                return false;
+                            })
```

Comment:
```suggestion
                            .filter(policy -> policy.getPermission().equals(READ_APPLICATIONS.getValue()))
```
Since we are only returning true/false, we can simplify the `filter` command. We can do this for all assertions in this test case.